tt_metal/impl/dispatch/command_queue_interface.hpp

```diff
@@ -177,7 +245,9 @@ inline uint32_t get_cq_issue_wr_ptr(chip_id_t chip_id, uint8_t cq_id, uint32_t c
     uint32_t recv;
     chip_id_t mmio_device_id = tt::Cluster::instance().get_associated_mmio_device(chip_id);
     uint16_t channel = tt::Cluster::instance().get_assigned_channel_for_device(chip_id);
-    tt::Cluster::instance().read_sysmem(&recv, sizeof(uint32_t), HOST_CQ_ISSUE_WRITE_PTR + get_relative_cq_offset(cq_id, cq_size), mmio_device_id, channel);
+    CoreType core_type = dispatch_core_manager::instance().get_dispatch_core_type(chip_id);
+    uint32_t issue_q_wr_ptr = dispatch_constants::get(core_type).get_host_command_queue_addr(CommandQueueHostAddrType::ISSUE_Q_WR);
```

Comment:
a lot of apis in here were updated to pull from dispatch_constants and get the core_type, I can cache these in sys mem manager to avoid further overheads

Reply:
results from pgm dispatch looks the same with and without these changes